tf2/src/buffer_core.cpp

```diff
@@ -528,23 +528,23 @@ tf2::TF2Error BufferCore::walkToTopParent(F& f, TimePoint time, CompactFrameID t
   if (frame_chain)
   {
     // Pruning: Compare the chains starting at the parent (end) until they differ
-    int m = (int)reverse_frame_chain.size()-1;
-    int n = (int)frame_chain->size()-1;
-    for (; m >= 0 && n >= 0; --m, --n)
+    size_t fcs = frame_chain->size() - 1;
+    size_t rcs = reverse_frame_chain.size() - 1;
```

Comment:
@ivanpauno what if either `.size()` is `0`? Same elsewhere.

Reply:
Thanks for catching this!
I modified the logic twice and I got confused.